firebase-dataconnect/connectors/connectors.gradle.kts

```diff
@@ -109,3 +110,41 @@ tasks.withType<KotlinCompile>().all {
     }
   }
 }
+
+// Adds a Gradle task that updates the JSON file that stores the list of Data Connect
+// executable versions.
+//
+// Example 1: Add versions 1.4.3 and 1.4.4 to the JSON file, and set 1.4.4 as the default:
+//   ../../gradlew -Pversions=1.4.3,1.4.4 -PdefaultVersion=1.4.4 updateJson --info
+//
+// Example 2: Add version 1.2.3 to the JSON file, but do not change the default version:
+//   ../../gradlew -Pversion=1.2.3 updateJson --info
+//
+// The `--info` argument can be omitted; it merely controls the level of log output.
+tasks.register<UpdateDataConnectExecutableVersionsTask>("updateJson") {
+  outputs.upToDateWhen { false }
+  jsonFile.set(project.layout.projectDirectory.file(
+    "../gradleplugin/plugin/src/main/resources/com/google/firebase/dataconnect/gradle/" +
+        "plugin/DataConnectExecutableVersions.json"))
+  workDirectory.set(project.layout.buildDirectory.dir("updateJson"))
+
+  val singleVersion: String? = project.providers.gradleProperty("version").orNull
+  val multipleVersions: List<String>? = project.providers.gradleProperty("versions").orNull?.split(',')
+  versions.set(buildList {
+    singleVersion?.let{add(it)}
+    multipleVersions?.let{addAll(it)}
+    if (isEmpty()) {
+      throw Exception("bm6d5ezxzd 'version' or 'versions' property must be specified")
+    }
+  })
+
+  updateMode.set(project.providers.gradleProperty("updateMode").map {
+    when (it) {
+      "overwrite" -> UpdateDataConnectExecutableVersionsTask.UpdateMode.Overwrite
+      "update" -> UpdateDataConnectExecutableVersionsTask.UpdateMode.Update
+      else -> throw Exception("ahe4zadcjs 'updateMode' must be 'overwrite' or 'update', but got: $it")
```

Comment:
They're just unique strings so that if you ever see that error you can easily search for it, like an error code. Gradle has particularly bad debugging from error messages so this can really help narrow down the line of code that threw the exception.

Reply:
I'll do that in a follow-up PR.

Reply:
Just an idea: in iOS we have something like this to narrow down the error message: https://github.com/firebase/firebase-ios-sdk/blob/main/FirebaseVertexAI/Sources/VertexLog.swift 
Maybe we can have something similar in Android.

Reply:
How does the SEO work with that? Like, what does an error message look like and how would you get back to the original source code line?

Reply:
The code will shows in debug log, like replacing [I-FDC000000]. You can not go back to the line it comes from, it is mainly used to debug error type
```
11.3.0-beta - [FirebaseDataConnect][I-FDC000000] DataConnect privacy logging disabled.
```